src/common/vector_operations/vector_cast.cpp

```diff
@@ -236,6 +240,7 @@ void VectorOperations::Cast(Vector &source, Vector &result, SQLType source_type,
 		timestamp_cast_switch(source, result, source_type, target_type, count);
 		break;
 	case SQLTypeId::VARCHAR:
+	case SQLTypeId::BLOB:
```

Comment:
Blob needs a separate switch here. VARCHAR -> TIMESTAMP is a valid cast, BLOB -> TIMESTAMP is not.